### PR TITLE
Update CHANGE_LOG for 0.37.0

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

File: CHANGE_LOG [hunk 1]
```diff
@@ -3,8 +3,8 @@ v0.38.0
 
 In development
 
-v0.37.0 RC2 (July 16 2021)
---------------------------
+v0.37.0 (August 18, 2021)
```
Comment (stuartarchibald) [inline CHANGE_LOG:6]: Note: This may need changing.

Comment (sklam) [reply CHANGE_LOG:6]:
```suggestion
v0.37.0 (August 19, 2021)
```

**Review by sklam — APPROVED** [review]
LGTM

**Comment by sklam** [comment]
RTD passed. No need to wait for Azure.

**Pull request merged**